libbeat/autodiscover/autodiscover.go

```diff
@@ -207,26 +207,25 @@ func (a *Autodiscover) handleStart(event bus.Event) bool {
 			continue
 		}
 
-		err = a.factory.CheckConfig(config)
-		if err != nil {
-			a.logger.Error(errors.Wrap(err, fmt.Sprintf(
-				"Auto discover config check failed for config '%s', won't start runner",
-				common.DebugString(config, true))))
-			continue
-		}
-
 		// Update meta no matter what
 		dynFields := a.meta.Store(hash, meta)
 
 		if cfg, ok := a.configs[eventID][hash]; ok {
 			a.logger.Debugf("Config %v is already running", common.DebugString(config, true))
 			newCfg[hash] = cfg
 			continue
-		} else {
-			newCfg[hash] = &reload.ConfigWithMeta{
-				Config: config,
-				Meta:   &dynFields,
-			}
+		}
+
+		err = a.factory.CheckConfig(config)
```

Comment:
Code looks good, but can we add a test?

Reply:
Test added, and with it I found a case I wasn't handling :slightly_smiling_face: when the adapter generates for the same event two equivalent configs. This is handled now checking if the hash exists in `newCfg`.

The autodiscover code is a bit tricky to test, I have added a channel so tests can listen when the worker has handled an event. I find this a bit hacky, but I didn't want to refactor more things now, and I couldn't find another reliable way of checking that _nothing_ happens when repeated configs are generated.

@ChrsMark let me know what you think.

Reply:
Nice catch with the duplicates in the same event! I think the test is valid since at least it covers the case we are dealing with here and should protect us in the future from failing to catch such cases in a possible refactoring.